documentation/src/main/asciidoc/userguide/chapters/domain/associations.adoc

```diff
@@ -485,3 +485,682 @@ Hibernate is not going to throw any exception, and it will assign a value of `nu
 include::{sourcedir}/NotFoundTest.java[tags=associations-not-found-non-existing-find-example,indent=0]
 ----
 ====
+
+[[associations-any]]
+==== `@Any` mapping
+
+The `@Any` mapping is useful to emulate a unidirectional `@ManyToOne` association when there can be multiple target entities.
+
+Because the `@Any` mapping defines a polymorphic association to classes from multiple tables,
+this association type requires the FK column which provides the associated parent identifier and
+a metadata information for the associated entity type.
+
+[NOTE]
+====
+It is impossible to specify a foreign key constraint for this kind of association.
+This is not the usual way of mapping polymorphic associations and you should use this only in special cases (e.g. audit logs, user session data, etc).
+====
+
+The `@Any` annotation describes the column holding the metadata information.
+To link the value of the metadata information and an actual entity type, the `@AnyDef` and `@AnyDefs` annotations are used.
+The `metaType` attribute allows the application to specify a custom type that maps database column values to persistent classes that have identifier properties of the type specified by `idType`.
+You must specify the mapping from values of the `metaType` to class names.
+
+For the next examples, consider the following `Property` class hierarchy:
+
+[[associations-any-property-example]]
+.`Property` class hierarchy
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/any/Property.java[tags=associations-any-property-example]
+
+include::{sourcedir}/any/IntegerProperty.java[tags=associations-any-property-example]
+
+include::{sourcedir}/any/StringProperty.java[tags=associations-any-property-example]
+----
+====
+
+A `PropertyHolder` can reference any such property, and, because each `Property` belongs to a separate table, the `@Any` annotation is, therefore, required.
+
+[[associations-any-example]]
+.`@Any` mapping usage
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/any/PropertyHolder.java[tags=associations-any-example]
+----
+
+[source, SQL, indent=0]
+----
+include::{extrasdir}/associations-any-example.sql[]
+----
+====
+
+As you can see, there are two columns used to reference a `Property` instance: `property_id` and `property_type`.
+The `property_id` is used to match the `id` column of either the `string_property` or `integer_property` tables,
+while the `property_type` is used to match the `string_property` or the  `integer_property` table.
```

Comment:
Very minor: 2 spaces here: "the  `integer_property`"

Reply:
Fixed.